arangod/RestHandler/RestBaseHandler.cpp

```diff
@@ -49,9 +47,20 @@ RestBaseHandler::RestBaseHandler(application_features::ApplicationServer& server
 ////////////////////////////////////////////////////////////////////////////////
 
 arangodb::velocypack::Slice RestBaseHandler::parseVPackBody(bool& success) {
+  bool strictValidation = true;
+  if (!_request->header(StaticStrings::ClusterCommSource).empty()) {
+    // cluster-internal request
+    AuthenticationFeature* af = AuthenticationFeature::instance();
+    if (!af->isActive() || _request->user().empty()) {
+      // with authentication off or by super user. that means the caller either
+      // has full privileges anyway, or is the superuser, which is trusted.
+      strictValidation = false;
+    }
+  }
+
   try {
     success = true;
-    return _request->payload(true);
+    return _request->payload(strictValidation);
```

Comment:
This should have been removed from the PR, as discussed.

Reply:
Correct, thanks for pointing this out agin.
Is now removed.